src/StarCluster/star_cluster.cpp

```diff
@@ -0,0 +1,249 @@
+		AMREX_ASSERT(!std::isnan(rho));
+
+		double vx = dvx(i, j, k);
+		double vy = dvy(i, j, k);
+		double vz = dvz(i, j, k);
+
+		vx *= renorm_amp * (rho / rho_max);
+		vy *= renorm_amp * (rho / rho_max);
+		vz *= renorm_amp * (rho / rho_max);
+
+		state_cc(i, j, k, HydroSystem<StarCluster>::density_index) = rho;
+		state_cc(i, j, k, HydroSystem<StarCluster>::x1Momentum_index) = rho * vx;
+		state_cc(i, j, k, HydroSystem<StarCluster>::x2Momentum_index) = rho * vy;
+		state_cc(i, j, k, HydroSystem<StarCluster>::x3Momentum_index) = rho * vz;
+		state_cc(i, j, k, HydroSystem<StarCluster>::energy_index) = 0;
+		state_cc(i, j, k, HydroSystem<StarCluster>::internalEnergy_index) = 0;
+	});
+}
+
+template <> void RadhydroSimulation<StarCluster>::ErrorEst(int lev, amrex::TagBoxArray &tags, amrex::Real /*time*/, int /*ngrow*/)
+{
+	// refine on Jeans length
+	const int N_cells = 4; // inverse of the 'Jeans number' [Truelove et al. (1997)]
+	const amrex::Real cs = quokka::EOS_Traits<StarCluster>::cs_isothermal;
+	const amrex::Real dx = geom[lev].CellSizeArray()[0];
+	const amrex::Real G = Gconst_;
+
+	auto const &state = state_new_cc_[lev].const_arrays();
+	auto tag = tags.arrays();
+
+	amrex::ParallelFor(tags, [=] AMREX_GPU_DEVICE(int bx, int i, int j, int k) noexcept {
+		Real const rho = state[bx](i, j, k, HydroSystem<StarCluster>::density_index);
+		const amrex::Real l_Jeans = cs * std::sqrt(M_PI / (G * rho));
+
+		if (l_Jeans < (N_cells * dx)) {
+			tag[bx](i, j, k) = amrex::TagBox::SET;
+		}
+	});
+}
+
+template <> void RadhydroSimulation<StarCluster>::ComputeDerivedVar(int lev, std::string const &dname, amrex::MultiFab &mf, const int ncomp_cc_in) const
+{
+	// compute derived variables and save in 'mf'
+	if (dname == "log_density") {
+		const int ncomp = ncomp_cc_in;
+		auto const &state = state_new_cc_[lev].const_arrays();
+		auto output = mf.arrays();
+
+		amrex::ParallelFor(mf, [=] AMREX_GPU_DEVICE(int bx, int i, int j, int k) noexcept {
+			Real const rho = state[bx](i, j, k, HydroSystem<StarCluster>::density_index);
+			output[bx](i, j, k, ncomp) = std::log10(rho);
+		});
+	}
+}
+
+auto problem_main() -> int
+{
+	// read problem parameters
+	amrex::ParmParse pp("perturb");
+
+	// cloud radius
+	Real R_sphere{};
+	pp.query("cloud_radius", R_sphere);
+
+	// cloud density
+	Real rho_sphere{};
+	pp.query("cloud_density", rho_sphere);
+
+	// cloud virial parameter
+	Real alpha_vir{};
+	pp.query("virial_parameter", alpha_vir);
+
+	// boundary conditions
+	const int ncomp_cc = Physics_Indices<StarCluster>::nvarTotal_cc;
+	amrex::Vector<amrex::BCRec> BCs_cc(ncomp_cc);
+	for (int n = 0; n < ncomp_cc; ++n) {
+		for (int i = 0; i < AMREX_SPACEDIM; ++i) {
+			BCs_cc[n].setLo(i, amrex::BCType::foextrap);
+			BCs_cc[n].setHi(i, amrex::BCType::foextrap);
+		}
+	}
+
+	// Problem initialization
+	RadhydroSimulation<StarCluster> sim(BCs_cc);
+	sim.doPoissonSolve_ = 1; // enable self-gravity
+	sim.Gconst_ = 1.0;	 // units where G = 1
```

Comment:
We can set this in the *.in file now, since you added that as a runtime parameter.

Reply:
I see you added that already. So this line can be deleted, since it's set from the params file instead.